### PR TITLE
fix: make sure old query stream doesn't block on close

### PR DESCRIPTION
**Pull request by purplefox**
### Description 

Fixes https://github.com/confluentinc/ksql/issues/5724

The ksqlDB CLI currently uses the old query streaming API at /query.

When terminating a query from the CLI, the connection is closed. The close is detected in the QueryStreamWriter poll loop which causes it to flush the OutputStream. This causes a write to be called on the underlying ResponseOutputStream, but because the channel isn't available for writing writeQueueFull() returns true and the thread is blocked until it times out (previously 60 seconds). This means that, previously, if you start and terminate queries quite rapidly, it takes 60 seconds for each ones state to be cleared up on the server. 

This PR changes the blocking in ResponseOutputStream so it can return more quickly if the stream is closed. It also changes the blocking timeout to 10 minutes, and it changes the server idle timeout to 10 minutes.

The old query endpoint had been migrated as is from Jetty to Vert.x. The old code is somewhat complex and this bug arose in the mapping from the synchronous approach taken in the old implementation and the Vert.x asynchronous APIs.

I'd recommend that the ksqlDB is refactored to use the new query streaming API (e.g. via the new Java client) which does not have these issues, and the old /query API is deprecated.

### Testing done 

Non functional change.

Manually verified.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: config/ksql-server.properties [hunk 1]
```diff
@@ -64,11 +64,8 @@ bootstrap.servers=localhost:9092
 compression.type=snappy
 
 #------ Connect -------
-
```
Comment (vcrfxia) [inline config/ksql-server.properties:67]: nit: unnecessary whitespace changes

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java [hunk 2]
```diff
@@ -113,19 +114,35 @@ static void handleOldApiResponse(final Server server, final RoutingContext routi
...
+        closeOutputStream(ros, promise);
       }
     }, vcf);
   }
 
+  private static void closeOutputStream(final OutputStream os, final Promise<Void> promise) {
+    try {
+      os.close();
+    } catch (IOException e) {
+      promise.fail(e);
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java:142]: Should we only try to fail the promise if it hasn't already been completed?

**Comment by apurvam** [comment]
Yes, let's target 6.0.x and merge.
